pkgs/applications/networking/instant-messengers/pybitmessage/default.nix

```diff
@@ -1,33 +1,29 @@
 { stdenv, fetchFromGitHub, pythonPackages, openssl }:
 
-stdenv.mkDerivation rec {
+pythonPackages.buildPythonApplication rec {
   name = "pybitmessage-${version}";
 
-  version = "0.4.4";
+  version = "0.6.2";
 
   src = fetchFromGitHub {
     owner = "bitmessage";
     repo = "PyBitmessage";
     rev = "v${version}";
-    sha256 = "1f4h0yc1mfjnxzvxiv9hxgak59mgr3a5ykv50vlyiay82za20jax";
+    sha256 = "04sgns9qczzw2152gqdr6bjyy4fmgs26cz8n3qck94l0j51rxhz8";
   };
 
-  buildInputs = with pythonPackages; [ python pyqt4 wrapPython ] ++ [ openssl ];
+  propagatedBuildInputs = with pythonPackages; [ msgpack pyqt4 ] ++ [ openssl ];
 
   preConfigure = ''
-    substituteInPlace Makefile \
-      --replace "PREFIX?=/usr/local" "" \
-      --replace "/usr" ""
-  '';
+    substituteInPlace setup.py \
+      --replace "nothing = raw_input()" pass \
+      --replace 'print "It looks like building the package failed.\n" \' pass \
+      --replace '    "You may be missing a C++ compiler and the OpenSSL headers."' pass
 
-  makeFlags = [ "DESTDIR=$(out)" ];
+  '';
 
-  postInstall = ''
-    substituteInPlace $out/bin/pybitmessage \
-      --replace "exec python2" "exec ${pythonPackages.python}/bin/python" \
-      --replace "/opt/openssl-compat-bitcoin/lib/" "${openssl.out}/lib/"
-    wrapProgram $out/bin/pybitmessage \
-      --prefix PYTHONPATH : "$(toPythonPath $out):$PYTHONPATH"
+  makeWrapperArgs = ''
```

Comment:
Patching the source to hardcode paths to libraries has preference over setting `LD_LIBRARY_PATH`.